multi/imxrt-multi/config.h

```diff
@@ -766,6 +766,54 @@
 #endif
 #endif
 
+#if defined(UART_CONSOLE)
+#if ISEMPTY(UART_CONSOLE)
+#error "UART_CONSOLE must not be empty"
+#elif UART_CONSOLE <= 0
+#error "Invalid value for UART_CONSOLE"
+#endif
+#endif
```

Comment:
`UART_CONSOLE` is a bit flaky - it may be used somewhere else (e.g. kernel, plo, some usermode code...) where it is expected to have a specific value and at the same time it serves as a catch-all for ease of use (just one define to set up UART in plo, kernel, and multidrv). I think you can relax restrictions on `UART_CONSOLE` and rely on `UART_CONSOLE_USER` (used on a few targets) to make the process of using consoles/UARTs less painful.

```suggestion
#ifndef UART_CONSOLE_USER
#define UART_CONSOLE_USER UART_CONSOLE
#endif
```
This needs the condition above (`#if !defined(UART_CONSOLE) && !defined(RTT_CHANNEL_CONSOLE)`) to be reverted back to `#ifndef UART_CONSOLE`.

Also see the comment below